src/proj_functions.jl

```diff
@@ -0,0 +1,211 @@
+@doc "Return true if the projection is a geographic coordinate system" ->
+is_latlong(proj::Projection) = _is_latlong(proj.rep)
+
+@doc "Return true if the projection is a geocentric coordinate system" ->
+is_geocent(proj::Projection) = _is_geocent(proj.rep)
+
+@doc "Return true if the projection is a geocentric coordinate system" ->
+is_identical(p1::Projection, p2::Projection) = _compare_datums(p1.rep, p2.rep)
```

Comment:
I'm not an expert on this stuff, but my impression is that `pj_compare_datum` only compares a very specific part of the projection definition: the "datum" ie, the underlying latlong coordinate system.

Unfortunately proj doesn't seem to provide a way to fully compare projections - see the comment at the top of `pj_transform`: https://github.com/OSGeo/proj.4/blob/master/src/pj_transform.c#L77